turbopack/crates/turbopack-core/src/source_map/utils.rs

```diff
@@ -21,3 +28,166 @@ pub fn add_default_ignore_list(map: &mut SourceMap) {
         map.add_to_ignore_list(ignored_id as _);
     }
 }
+
+#[derive(Serialize, Deserialize)]
+struct SourceMapSectionOffsetJson {
+    line: u32,
+    offset: u32,
+}
+
+#[derive(Serialize, Deserialize)]
+struct SourceMapSectionItemJson {
+    offset: SourceMapSectionOffsetJson,
+    map: SourceMapJson,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct SourceMapJson {
+    version: u32,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    file: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    source_root: Option<String>,
+    // Technically a required field, but we don't want to error here.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    sources: Option<Vec<Option<String>>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    sources_content: Option<Vec<Option<String>>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    names: Option<Vec<String>>,
+    mappings: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    ignore_list: Option<Vec<u32>>,
+
+    // A somewhat widespread non-standard extension
+    debug_id: Option<String>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    sections: Option<Vec<SourceMapSectionItemJson>>,
+}
```

Comment:
Instead of demanding this strict structure from the source map, you could deserialize it as `serde_json::Value` and only pick the fields you are interested in and deserialize them with `serde_json::from_value`.